vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/spec/VssNodePropertiesBuilder.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2023 - 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa.vssprocessor.spec
+
+import org.eclipse.kuksa.vssprocessor.parser.KEY_DATA_COMMENT
+import org.eclipse.kuksa.vssprocessor.parser.KEY_DATA_DATATYPE
+import org.eclipse.kuksa.vssprocessor.parser.KEY_DATA_DESCRIPTION
+import org.eclipse.kuksa.vssprocessor.parser.KEY_DATA_MAX
+import org.eclipse.kuksa.vssprocessor.parser.KEY_DATA_MIN
+import org.eclipse.kuksa.vssprocessor.parser.KEY_DATA_TYPE
+import org.eclipse.kuksa.vssprocessor.parser.KEY_DATA_UNIT
+import org.eclipse.kuksa.vssprocessor.parser.KEY_DATA_UUID
+import kotlin.reflect.KClass
+
+internal class VssNodePropertiesBuilder(
```

Comment:
Much more straightforward to use with a builder now <3

I still think there is room for improvement e.g. since we now introduced a builder should we maybe directly create a VssNodeSpec (which has a property for each attribute of the VSS) instead of just returning a Set<VssNodeProperty). 

In the end we will have to keep the set anyway to make it easier to iterate over, so I left it as is but am still open for suggestions here

Reply:
Let us shift further improvements to the point when we are doing the VssNodeSpec refactoring.